.github/workflows/zxc-single-day-longevity-test.yaml

```
@@ -544,7 +565,7 @@ jobs:
   longevity-tests-finish:
     name: Read results of Longevity test
     runs-on: hiero-citr-linux-large
-    if: always()
+    if: ${{ !cancelled() }}
```

```suggestion
    if: ${{ !cancelled() && always() }}
```

I think you need to specify `&& always()` here to tell the step to run if previous steps failed.

---

adding always() forces cancelled run to dice into unnecessary steps, as I tested.